src/Features/CSharp/Portable/CodeFixes/QualifyMemberAccess/CSharpQualifyMemberAccessCodeFixProvider.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CodeFixes.Qualify;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeFixes.QualifyMemberAccess
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.QualifyMemberAccess), Shared]
+    [ExtensionOrder(After = PredefinedCodeFixProviderNames.RemoveUnnecessaryCast)]
+    internal class CSharpQualifyMemberAccessCodeFixProvider : AbstractQualifyMemberAccessCodeFixprovider<SimpleNameSyntax>
+    {
+        protected override string GetId() => CSharpFeaturesResources.AddThisQualification;
+
+        protected override string GetTitle() => CSharpFeaturesResources.AddThisQualification;
+
+        protected override SyntaxNode GetReplacement(SimpleNameSyntax node)
```

Comment:
Good catch.  I've updated it to use the syntax generator.  Once I fix my merge conflicts after @jasonmalinowski's huge change, I'll update this PR.